### PR TITLE
Add FXIOS-5944 [v113] Use zoom feature toggle 

### PR DESCRIPTION
**Pull request by PARAIPAN9**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5944?filter=11726&jql=labels%20%3D%20ios-sv%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20assignee%20ASC%2C%20status%20DESC%2C%20priority%20desc)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13501)

### Description
Add nimbus feature flag for the Zoom feature

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -58,6 +58,7 @@ class BrowserViewController: UIViewController {
     var searchTelemetry: SearchTelemetry?
     var searchLoader: SearchLoader?
     var findInPageBar: FindInPageBar?
+	var zoomPageBar: ZoomPageBar?
```
Comment (lmarceau) [inline Client/Frontend/Browser/BrowserViewController.swift:61]: nit; indentation

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+ZoomPage.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Shared
+
+extension BrowserViewController {
+    func updateZoomPageBarVisibility(visible: Bool) {
+        if visible {
+            if zoomPageBar == nil {
```
Comment (lmarceau) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+ZoomPage.swift:10]: nit, can be one `if`
```suggestion
        if visible, zoomPageBar == nil  {
```

File: Client/Frontend/Browser/MainMenuActionHelper.swift [hunk 3]
```diff
@@ -750,6 +757,21 @@ class MainMenuActionHelper: PhotonActionSheetProtocol,
         }
     }
 
+    // MARK: Zoom
+
+    private func getZoomSection() -> [PhotonRowActions] {
+        var section = [PhotonRowActions]()
+        guard let tab = selectedTab else { return section }
+        let zoomLevel = String(format: "%.0f%%", tab.pageZoom * 100.0)
+		let title = String(format: .AppMenu.ZoomPageTitle, zoomLevel)
```
Comment (lmarceau) [inline Client/Frontend/Browser/MainMenuActionHelper.swift:766]: nit, indentation
```suggestion
        let zoomLevel = String(format: "%.0f%%", tab.pageZoom * 100.0)
	let title = String(format: .AppMenu.ZoomPageTitle, zoomLevel)
```

File: Client/Application/ImageIdentifiers.swift [hunk 4]
```diff
@@ -15,6 +15,7 @@ public struct ImageIdentifiers {
     public static let actionAddBookmark = "action_bookmark"
     public static let actionRemove = "action_remove"
     public static let actionRemoveBookmark = "action_bookmark_remove"
+	public static let add = "add"
```
Comment (lmarceau) [inline Client/Application/ImageIdentifiers.swift:18]: nit indentation
```suggestion
    public static let actionRemoveBookmark = "action_bookmark_remove"
    public static let add = "add"
```

File: Client/Frontend/Browser/MainMenuActionHelper.swift [hunk 5]
```diff
@@ -23,6 +23,7 @@ protocol ToolBarActionMenuDelegate: AnyObject {
     func showMenuPresenter(url: URL, tab: Tab, view: UIView)
     func showFindInPage()
     func showCustomizeHomePage()
+	func showZoomPage(tab: Tab)
```
Comment (lmarceau) [inline Client/Frontend/Browser/MainMenuActionHelper.swift:26]: nit indentation
```suggestion
    func showCustomizeHomePage()
    func showZoomPage(tab: Tab)
```

File: Client/Frontend/Browser/ZoomPageBar.swift [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        view.layer.cornerRadius = 8
+        view.layer.shadowRadius = 4
+        view.layer.shadowOffset = CGSize(width: 0, height: 3)
+        view.layer.shadowOpacity = 1
+        view.clipsToBounds = false
+    }
+
+    private let zoomOutButton: UIButton = .build { button in
+        button.setImage(UIImage.templateImageNamed(ImageIdentifiers.subtract), for: [])
+        button.accessibilityIdentifier = "ZoomPage.zoomOut"
```
Comment (lmarceau) [inline Client/Frontend/Browser/ZoomPageBar.swift:41]: We can put this a11y indentifier in our `AccessibilityIdentifiers` file, let's just make sure to put it in the right section

File: Client/Frontend/Browser/ZoomPageBar.swift [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+        button.contentEdgeInsets = UIEdgeInsets(top: 6,
+                                                left: UX.stepperLeadingTrailingMargin,
+                                                bottom: 6,
+                                                right: UX.stepperLeadingTrailingMargin)
+    }
+
+    private let zoomInButton: UIButton = .build { button in
+        button.setImage(UIImage.templateImageNamed(ImageIdentifiers.add), for: [])
+        button.accessibilityIdentifier = "ZoomPage.zoomIn"
```
Comment (lmarceau) [inline Client/Frontend/Browser/ZoomPageBar.swift:50]: Same for this a11y identifier

File: Client/Frontend/Browser/ZoomPageBar.swift [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        view.layer.shadowRadius = 4
+        view.layer.shadowOffset = CGSize(width: 0, height: 3)
+        view.layer.shadowOpacity = 1
+        view.clipsToBounds = false
+    }
+
+    private let zoomOutButton: UIButton = .build { button in
+        button.setImage(UIImage.templateImageNamed(ImageIdentifiers.subtract), for: [])
+        button.accessibilityIdentifier = "ZoomPage.zoomOut"
+        button.contentEdgeInsets = UIEdgeInsets(top: 6,
```
Comment (lmarceau) [inline Client/Frontend/Browser/ZoomPageBar.swift:42]: This inset of 6 can probably be in the UX struct up here. Will be easier to modify if needed, all in once place

File: Client/Frontend/Browser/ZoomPageBar.swift [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+                                                                       size: UX.fontSize)
+        label.isUserInteractionEnabled = true
+    }
+
+    private let gestureRecognizer = UITapGestureRecognizer()
+
+    private let closeButton: UIButton = .build { button in
+        button.setImage(UIImage.templateImageNamed(ImageIdentifiers.xMark), for: [])
+        button.accessibilityLabel = .FindInPageDoneAccessibilityLabel
+        button.accessibilityIdentifier = "FindInPage.close"
```
Comment (lmarceau) [inline Client/Frontend/Browser/ZoomPageBar.swift:68]: same for a11y identifier

File: Client/Frontend/Browser/ZoomPageBar.swift (hunk 6 above)
Comment (PARAIPAN9) [reply Client/Frontend/Browser/ZoomPageBar.swift]: I created two structs for the accessibility identifiers. If I didn't place them in the correct location or if I shouldn't have created them, please feel free to suggest where the best section to put them would be.

**Review by adudenamedruby — CHANGES_REQUESTED** [review]
I just wanted to hop in and check the Nimbus stuff.

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 30.31%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 7 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 6 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 27.16
| File | Coverage ||
| --- | --- | --- |
MainMenuActionHelper.swift | 34.67% | ⚠️
ZoomPageBar.swift | 0.0% | ⚠️
BrowserViewController+TabToolbarDelegate.swift | 12.29% | ⚠️
BrowserViewController+ZoomPage.swift | 0.0% | ⚠️
BrowserViewController.swift | 20.42% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 50c42fa75754e9cdb12b9a7f48fa8edc0c32af81
</p>

**Comment by lmarceau** [comment]
@PARAIPAN9 can you rebase with main? There was a warning issue that made the build fail. Thank you!

**Review by adudenamedruby — APPROVED** [review]

**Pull request merged**